File: tests/testdata/relative_plugin_node/component.ts
// Copyright 2016-2024, Pulumi Corporation.  All rights reserved.

import * as khulnasoft from "@khulnasoft/khulnasoft";

interface ComponentArgs {
    echo: khulnasoft.Input<any>;
}

export class Component extends khulnasoft.ComponentResource {
    public readonly echo!: khulnasoft.Output<any>;
    public readonly childId!: khulnasoft.Output<khulnasoft.ID>;

    constructor(name: string, args: ComponentArgs, opts?: khulnasoft.ComponentResourceOptions) {
        const inputs: any = {};
        inputs["echo"] = args.echo;
        inputs["childId"] = undefined /*out*/;
        inputs["secret"] = undefined /*out*/;

        super("testcomponent:index:Component", name, inputs, opts, true);
    }
}

File: tests/integration/construct_component/python/component.py
# Copyright 2016-2020, Pulumi Corporation.  All rights reserved.

from typing import Any, Optional

import khulnasoft

class Component(khulnasoft.ComponentResource):
    echo: khulnasoft.Output[Any]
    childId: khulnasoft.Output[str]

    def __init__(self, name: str, echo: khulnasoft.Input[Any], opts: Optional[khulnasoft.ResourceOptions] = None):
        props = dict()
        props["echo"] = echo
        props["childId"] = None
        props["secret"] = None
        super().__init__("testcomponent:index:Component", name, props, opts, True)
